### PR TITLE
Fix #589: Fix aclocal.m4 grep under SunOS / OpenIndiana

### LINKED ISSUE #589: sunos: ./configure -> grep: illegal option -- o
./configure on sunos 5.11:

grep: illegal option -- o
Usage: grep [-c|-l|-q] -bhinsvw pattern file . . .

analysis:

eggdrops usage of grep is not portable, gnu grep works, sunos grep not

aclocal.m4:

TCL_PTHREAD_LDFLAG=`echo $TCL_EXTRA_CFLAGS | grep -o -- '-pthread'`

eggdrop compiles and runs anyway, but the grep error in configures output is annoying enough

### PR DESCRIPTION
**Pull request by michaelortmann**
Found by: michaelortmann
Patch by: michaelortmann
Fixes: #589

One-line summary:
Fix #589: Fix aclocal.m4 grep under SunOS / OpenIndiana

Additional description (if needed):
Also update BUG-REPORT and COMPILE-GUIDE for SunOS / OpenIndiana

Test cases demonstrating functionality (if applicable):

1. SunOS / OpenIndiana 5.11:

before:
```
$ ./configure
[...]
grep: illegal option -- o
usage:  grep [-E|-F] [-bchHilnqrRsvx] [-A num] [-B num] [-C num|-num]
             [-e pattern_list]... [-f pattern_file]... [pattern_list] [file]...
[...]
```

after:

```
$ ./configure
[...]
checking for Tcl library flags...  -L/usr/lib -ltcl8.6 -ldl -lz  -lsocket -lnsl -lpthread -lm
[...]
```

-pthread was not found via grep in $TCL_EXTRA_CFLAGS, ok

2. OpenBSD 6.3:

This is a test case for another operating system to demonstrate, the Fix shows no regression there

```
$ ./configure
[...]
checking for Tcl library flags... -pthread -L/usr/local/lib -ltcl86  -lz  -lm
[...]
```

-pthread was found via grep in $TCL_EXTRA_CFLAGS, ok

3. Other and/oder older non-gnu-grep Systems also show show this bug fixed with this patch

Example: OpenBSD 4.9:

grep: unknown option -- o
usage: grep [-abcEFGhIiLlnqRsUVvwxZ] [-A num] [-B num] [-C[num]]
        [-e pattern] [-f file] [--binary-files=value] [--context[=num]]
        [--line-buffered] [pattern] [file ...]

### CODE REVIEW

File: aclocal.m4 [hunk 1]
```diff
@@ -1040,7 +1040,11 @@ AC_DEFUN([EGG_TCL_TCLCONFIG],
     if test -r ${TCL_BIN_DIR}/tclConfig.sh; then
       . ${TCL_BIN_DIR}/tclConfig.sh
       # OpenBSD uses -pthread, but tclConfig.sh provides that flag in EXTRA_CFLAGS
-      TCL_PTHREAD_LDFLAG=`echo $TCL_EXTRA_CFLAGS | grep -o -- '-pthread'`
+      if [[ "$(echo $TCL_EXTRA_CFLAGS | grep -- "-pthread")" != "" ]]; then
```
Comment (Cizzle) [inline aclocal.m4]: Don't use the [[ bashism but `test`. Would prefer for conformity:
```
TCL_PTHREAD_LDFLAG=`echo $TCL_EXTRA_CFLAGS | grep -- '-pthread'`
if test "x$TCL_PTHREAD_LDFLAG" != x; then
  TCL_PTHREAD_LDFLAG="-pthread"
fi
```
Same for below.

Comment (michaelortmann) [reply aclocal.m4]: fixed, tested again, ok.

**Comment by Cizzle** [comment]
LGTM

**Comment by vanosg** [comment]
build

**Comment by vanosg** [comment]
build

**Pull request merged**